components/google_drive/actions/add-file-sharing-preference/add-file-sharing-preference.js

```diff
@@ -0,0 +1,96 @@
+const googleDrive = require("../../google_drive.app");
+
+module.exports = {
+  key: "google_drive-add-file-sharing-preference",
+  name: "Add File Sharing Preference",
+  description: "Add File Sharing Preference",
+  version: "0.0.12",
+  type: "action",
+  props: {
+    googleDrive,
+    drive: {
+      propDefinition: [
+        googleDrive,
+        "watchedDrive",
+      ],
+      description: "The drive you want to find a file in",
+      optional: true,
+      default: "",
+    },
+    fileId: {
+      propDefinition: [
+        googleDrive,
+        "fileId",
+        (c) => ({
+          drive: c.drive,
+          baseOpts: {
+            q: "'me' in owners",
+          },
+        }),
+      ],
+      optional: false,
+      description: "The file to add a file sharing preference to.",
+    },
+    role: {
+      type: "string",
+      label: "Role",
+      description: "The role granted by this permission.",
+      optional: true,
+      default: "reader",
+      options: [
+        "owner",
+        "organizer",
+        "fileOrganizer",
+        "writer",
+        "commenter",
+        "reader",
+      ],
+    },
+    type: {
+      type: "string",
+      label: "Type",
+      description: "The type of the grantee.",
+      optional: true,
+      default: "anyone",
+      options: [
+        "user",
+        "group",
+        "domain",
+        "anyone",
+      ],
+    },
+    domain: {
+      type: "string",
+      label: "Domain",
+      description:
+        "The domain to which this permission refers if type is 'domain'.",
+      optional: true,
+      default: "",
```

Comment:
Just to confirm, is `""` the correct default for this param, or `undefined`?

Reply:
`undefined` is the correct default for this param.

Reply:
Thanks, you can remove completely then (see prior comment)